src/test/index.spec.ts

```diff
@@ -1955,4 +2053,144 @@ test.suite('ts-node', (test) => {
       });
     }
   });
+
+  test.suite('top level await', (test) => {
+    const compilerOptions = {
+      target: semver.gte(ts.version, '3.0.1')
+        ? 'es2018'
+        : // TODO should this be kept if tla is not supported on TS <3.8.0?
+          // TS 2.7 is using polyfill for async interator even though they
+          // were added in es2018
+          'esnext',
+    };
+    function executeInTlaRepl(input: string, waitMs = 1000) {
+      return executeInRepl(
+        input
+          .split('\n')
+          .map((line) => line.trim())
+          .join(''),
+        {
+          waitMs,
+          createServiceOpts: {
+            experimentalReplAwait: true,
+            compilerOptions,
+          },
+          startOptions: { useGlobal: false },
+        }
+      );
+    }
+
+    if (semver.gte(ts.version, '3.8.0')) {
+      // Serial because it's timing-sensitive
+      test.serial('should allow evaluating top level await', async () => {
+        const script = `
+        export {}; /* TODO remove this line; implement automatic export{}; in the REPL */
+        const x: number = await new Promise((r) => r(1));
```

Comment:
was the `: number` necessary here?

Reply:
Yeah, when it's implied to by `unknown` then the tests fail because an unexpected diagnostic is raised.